### PR TITLE
Note about capping deployments to ~100 replicas

### LINKED ISSUE #1101: High API Calls quota consumption
**Describe the bug**

Hey folks,

Thanks for this awesome controller.

We recently had really high GitHub API quota consumption on our runner pool managed with this controller.

**Checks**

- [x] My actions-runner-controller version (v0.20.4) does support the feature
- [ ] I'm using an unreleased version of the controller I built from HEAD of the default branch

**To Reproduce**

When scaling to 100+ replicas, we're starting to see a very high amount of API calls. Not sure it's related tho; but our workflows didn't change (no rogue job looping on a github call).
- Could it be them, our "idle" runners, that are constantly polling the API ?
- Is there a known stability limit of # of replicas?
- Did we get anything wrong?

**Expected behavior**

We expected our runner pools to scale OK with the load, and were planning on adding another controller with a separate Github App to balance the API usage. Well, the wall came earlier.

**Screenshots**

Our workload is roughly the same:
image

When we hit the quota limit, well, we had a queuing spike:
image

But overall it was a "normal" day; except the huge spike in quota consumption: 
Capture d’écran 2022-02-08 à 18 39 53

You can notice at the right part of this graph above that we added more controllers with their own Github App (1 to 100 replicas each), trying to ease the pain.. But nope, both of them were consumed very quickly.

FTR, here is what a normal day (used to) look like : 
image

**Environment (please complete the following information):**
 - Controller Version: 0.20.4
 - Deployment Method: vanilla YAML apply
 - Running on EKS 1.20
 - Using Organization runners

**Additional context**

Our current setup is the following:

- 1 EKS clusters
- 3 actions-runner-controllers in 3 different namespaces (different labels are applied on the runners, depending on the controller).
- using the same runner groups (is it an (scaling) issue?)

### PR DESCRIPTION
**Pull request by Djiit**
Fixes #1101

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -327,6 +327,8 @@ Now you can see the runner on the enterprise level (if you have enterprise acces
 
 You can manage sets of runners instead of individually through the `RunnerDeployment` kind and its `replicas:` attribute. This kind is required for many of the advanced features.
 
+> It is recommended to have less than 100 replicas per deployment to avoid exhausting your GitHub API Quota.
```
Comment (toast-gear) [inline README.md:330]: * fewer than

**Comment by toast-gear** [comment]
Let me have a think about the best way to document this. In the future we want `Stateful Runners` to take a more prominent place in the docs and this will be relevant to them too.

**Comment by toast-gear** [comment]
I think it's probably best to put it as a secon `>` note after:

```
> Since the release of GitHub's [`workflow_job` webhook](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_job), webhook driven scaling is the preferred way of autoscaling as it enables targeted scaling of your `RunnerDeployment` / `RunnerSet` as it includes the `runs-on` information needed to scale the appropriate runners for that workflow run. More broadly, webhook driven scaling is the preferred scaling option as it is far quicker compared to the pull driven scaling and is easy to setup.
```

Additionally could could you change it to:

```
> If you are not using GHES, and so can't set your rate limit budget, it is recommended to have fewer than 100 replicas per deployment to avoid exhausting your budget.
```

Cheers pal

**Review by genisd — APPROVED** [review]
not that I have anything to say. as a symbolic gesture here is my approval :)

**Comment by toast-gear** [comment]
This hasn't been merged on purpose, it will get merge after the next release which is immediate. The next release includes caching of requests so this warning won't hold true. It will need to highlight this applies to controller version `< v0.22.0` in the style of the other version warnings:

```
> If you are using controller version => [v0.22.0](https://github.com/actions-runner-controller/actions-runner-controller/releases/tag/v0.22.0) and you are not using GHES, and so can't set your rate limit budget, it is recommended that you use 100 replicas or fewer to prevent being rate limited
```

Merging it now will just cause confusion and will need updating as soon as the next release is out.

**Comment by toast-gear** [comment]
@Djiit there were various changes to the readme for the latest release. If you could resolve the conflicts in this PR and change the message to be the below we can get this merged:

```
> If you are using controller version < [v0.22.0](https://github.com/actions-runner-controller/actions-runner-controller/releases/tag/v0.22.0) and you are not using GHES, and so can't set your rate limit budget, it is recommended that you use 100 replicas or fewer to prevent being rate limited
```

**Pull request merged**